java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/AzureEventHubsSourceTask.java

```diff
@@ -28,29 +45,51 @@ public String version() {
 
   @Override
   public void start(Map<String, String> props) {
-    log.info("Task initialising");
-    printAsciiHeader(jarManifest, "/azure-eventhubs-ascii.txt");
+    Optional<SourceTaskContext> contextOptional = ofNullable(context);
+    Map<String, String> configProps = null;
+    if (contextOptional.isPresent()) {
+      configProps = context.configs();
+      ofNullable(context.offsetStorageReader())
+          .ifPresent(TopicPartitionOffsetProvider::initialize);
```

Comment:
As mentioned, i think this does not add value creating an instance, deal with synchronisation just to have the reference on offsetStorageReader.